core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/GraphQLActionBuilder.kt

```diff
@@ -0,0 +1,1011 @@
+                    OptionalGene(element.typeName, DateGene(element.typeName))
+                else
+                    DateGene(element.typeName)
+            GqlConst.ENUM ->
+                return if (element.isKindOfFieldTypeOptional)
+                    OptionalGene(element.typeName, EnumGene(element.typeName, element.enumValues))
+                else
+                    EnumGene(element.typeName, element.enumValues)
+            GqlConst.SCALAR -> {
+                val copy = element.copy(
+                    fieldType = element.fieldType, KindOfFieldName = element.typeName,
+                    typeName = element.KindOfFieldName
+                )
+                return getInputGene(state, history, maxNumberOfGenes, copy)
+            }
+            "id" ->
+                return if (element.isKindOfFieldTypeOptional)
+                    OptionalGene(element.typeName, StringGene(element.typeName))
+                else
+                    StringGene(element.typeName)
+
+            GqlConst.UNION -> {
+                LoggingUtil.uniqueWarn(log, " GQL does not support union in input type: ${element.KindOfFieldName}")
+                return StringGene("Not supported type")
+            }
+            GqlConst.INTERFACE -> {
+                LoggingUtil.uniqueWarn(
+                    log,
+                    "GQL does not support interface in input type: ${element.KindOfFieldName}"
+                )
+                return StringGene("Not supported type")
+            }
+            else ->
+                return if (element.isKindOfFieldTypeOptional)
+                    OptionalGene(element.typeName, StringGene(element.typeName))
+                else
+                    StringGene(element.typeName)
+
+        }
+    }
+
+    /**
+     * Create input object gene
+     */
+    private fun createInputObjectGene(
+        state: TempState,
+        history: Deque<String>,
+        maxNumberOfGenes: Int,
+        element: Table
+    ): Gene {
+        val fields: MutableList<Gene> = mutableListOf()
+        val selectionInArgs = state.argsTablesIndexedByName[element.typeName] ?: listOf()
+        for (element in selectionInArgs) {
+            if (element.kindOfFieldType.lowercase() == GqlConst.SCALAR) {
+                val copy = element.copy(
+                    fieldType = element.typeName,
+                    KindOfFieldName = element.fieldType,
+                    kindOfFieldType = element.kindOfFieldType,
+                    typeName = element.fieldName,
+                    isKindOfFieldTypeOptional = element.isKindOfFieldTypeOptional,
+                    isKindOfFieldNameOptional = element.isKindOfFieldNameOptional,
+                    enumValues = element.enumValues,
+                    unionTypes = element.unionTypes,
+                    interfaceTypes = element.interfaceTypes,
+                    isFieldNameWithArgs = element.isFieldNameWithArgs,
+                )
```

Comment:
aren't some of these assignments on same variable names redundant? eg, `interfaceTypes = element.interfaceTypes`

Reply:
Done